tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -86,12 +86,37 @@ class RFieldBase {
    friend struct ROOT::Experimental::Internal::RFieldCallbackInjector; // used for unit tests
    using ReadCallback_t = std::function<void(void *)>;
 
+protected:
+   /// A functor to release the memory acquired by GenerateValue (memory and constructor).
+   /// This implementation works for types with a trivial destructor. More complex fields implement a derived deleter.
+   /// The deleter is operational without the field object and thus can be used to destruct/release a value after
+   /// the field has been destructed.
```

Comment:
IIUC this is really what distinguishes `RDeleter` from `RField::DestroyValue` and the main motivation to introduce this extra class. I think it would be nice to add a test that explicitly reproduces this case (if it's not there already)

Reply:
Unit test added.